@xen-orchestra/fs/src/s3.js

```diff
@@ -116,34 +137,50 @@ export default class S3Handler extends RemoteHandlerAbstract {
   }
 
   async _isNotEmptyDir(path) {
-    const result = await this._s3.listObjectsV2({
-      Bucket: this._bucket,
-      MaxKeys: 1,
-      Prefix: this._makePrefix(path),
-    })
-    return result.Contents.length !== 0
+    const result = await this._s3.send(
+      new ListObjectsV2Command({
+        Bucket: this._bucket,
+        MaxKeys: 1,
+        Prefix: this._makePrefix(path),
+      })
+    );
+    return result.Contents?.length > 0
   }
 
   async _isFile(path) {
     try {
-      await this._s3.headObject(this._createParams(path))
+      await this._s3.send(
+        new HeadObjectCommand(this._createParams(path))
+      );
       return true
     } catch (error) {
-      if (error.code === 'NotFound') {
+      if (error.$response.statusCode === 404) {
         return false
       }
       throw error
     }
   }
 
   async _outputStream(path, input, { validator }) {
-    await this._s3.upload(
-      {
+    const Body = new PassThrough();
+    pipeline(input, Body, () => {});
+
+    const upload = new Upload({
```

Comment:
Why not use `PutObjectCommand` instead?

See https://docs.aws.amazon.com/sdk-for-javascript/v3/developer-guide/javascript_s3_code_examples.html